cli/command/utils.go

```diff
@@ -115,5 +117,23 @@ func PruneFilters(dockerCli Cli, pruneFilters filters.Args) filters.Args {
 		pruneFilters.Add(parts[0], parts[1])
 	}
 
+	// Server APIs older than 1.29 will just ignore label filters.
+	// It's dangerous because a command like:
+	// `docker container prune --filter label=foo=bar`
+	// would just remove all stopped containers without considering the label
+	// The remote API should reject requests using unknown filters.
+	if pruneFilters.Get("label") != nil {
+		serverVersion, err := dockerCli.Client().ServerVersion(context.Background())
```

Comment:
@dnephin 
> // ClientVersion returns the version string associated with this instance of the Client

Are you sure? What's important is to detect the version server side.